### ISSUE TITLE
Expose the "stats" object to custom reports

### ISSUE THREAD

**Issue by exhuma** [PyCQA/pylint#1799]
I was trying to writ a custom report class to be used with the `-f` CLI argument.

The main goal was to create a parseable output format which includes overall stats to be able to calculate a score (like the score shown in the text output format). 

I ran into a roadblock as the statistics are not exposed to the reports class. They are interpolated into a simple line of text which gets forwarded as "Section" into the report.
    https://github.com/PyCQA/pylint/blob/df53e1eb10c72c77036ef22e59e1b81f6b96582b/pylint/utils.py#L826

So when implementing a custom report, I have no access to that object anymore. The crucial thing missing is the total number of processed statements. It is already possible to determine the other values by overriding the "handle_message" method.

**Comment by PCManticore** [comment]
Sounds like a good change to me.

**Comment by brycepg** [comment]
The `stats` object is available through the Linter class through the Run class:

```python
from pylint import lint
run = lint.Run([sys.argv[1]], do_exit=False)
print(run.linter.stats['statement'])
```

**Comment by PCManticore** [comment]
Thanks for the comment, Bryce! @exhuma This means you can already access the `stats` object using `self.linter.stats`, so no need for another change here.
